test/shared/src/main/scala/zio/test/Gen.scala

```diff
@@ -84,26 +136,73 @@ object Gen {
     Gen(ZStream.fromEffect(ZIO.accessM[Random](r => f(r.random))))
 
   /**
-   * A generator of integral values inside the specified range: [start, end).
+   * A generator of integers inside the specified range: [start, end].
+   * The shrinker will shrink toward the lower end of the range ("smallest").
+   */
+  final def int(range: Range): Gen[Random, Int] =
+    integral[Int](range)
+
+  /**
+   * A generator of integral values inside the specified range: [start, end].
    * The shrinker will shrink toward the lower end of the range ("smallest").
    */
   final def integral[A](range: Range)(implicit I: Integral[A]): Gen[Random, A] =
     fromEffectSample(
-      nextInt(range.end - range.start)
+      nextInt(range.end - range.start + 1)
         .map(r => I.fromInt(r + range.start))
-        .map(int => Sample(int, ZStream.fromIterable(range.drop(1).reverse).map(I.fromInt(_))))
+        .map(int => Sample(int, ZStream.fromIterable(range.start.until(I.toInt(int)).reverse).map(n => I.fromInt(n))))
     )
 
+  final def listOf[R, A](range: Range)(gen: Gen[R, A]): Gen[R with Random, List[A]] =
+    int(range).flatMap(n => listOfN(n)(gen))
+
+  final def listOfN[R, A](n: Int)(gen: Gen[R, A]): Gen[R, List[A]] =
+    List.fill(n)(gen).foldRight[Gen[R, List[A]]](const(Nil))((a, gen) => a.zipWith(gen)(_ :: _))
+
+  final def optionOf[R, A](gen: Gen[R, A]): Gen[R with Random, Option[A]] =
+    oneOf(const(None), gen.map(Some(_)))
+
+  final def oneOf[R, A](a: Gen[R, A], as: Gen[R, A]*): Gen[R with Random, A] =
+    int(0 to as.length).flatMap(n => if (n == 0) a else as(n - 1))
+
   /**
    * A sized generator, whose size falls within the specified bounds.
    */
   final def sized[R <: Random, A](min: Int, max: Int)(f: Int => Gen[R, A]): Gen[R, A] =
```

Comment:
Probably you can refactor the `min: Int, max: Int` to be a `Range` like the other combinators you introduced.

Reply:
Yes.